test/integration/protocol_integration_test.cc

```diff
@@ -757,7 +757,29 @@ TEST_P(ProtocolIntegrationTest, Retry) {
     for (auto& histogram : test_server_->histograms()) {
       if (histogram->name() == name) {
         const auto& stats = histogram->cumulativeStatistics();
-        return stats.sampleCount();
+
+        // Note: we need to read the sample count from the main thread, to avoid data races.
+        uint64_t sample_count = 0;
+        Thread::MutexBasicLockable mu;
+        Thread::CondVar cv;
```

Comment:
Consider using absl::Notification instead of a mutex and a condvar.

The code here would be something like:
```
uint64_t sample_count = 0;
absl::Notification notification;
test_server_->server().dispatcher().post([&] {
  sample_count = stats.sampleCount();
  notification.Notify();
});

notification.WaitForNotification();
```

Reply:
Updated, thanks.